### PR TITLE
DNSName: correct len and offset types

### PR DESCRIPTION
**Pull request by chbruyand**
### Short description
This PR ~format `dnsname.{hh,cc}` files and~ corrects the type of `len` and `offset` arguments from `int` to `size_t`.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [ ] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

**Comment by rgacogne** [comment]
I'm wondering if it would be better to move the formatting of the files to a different pull request, so that we can focus on the (hopefully) non-impact of the type changes in this PR? In theory it should not matter but we get clang-tidy warnings for the whole PR, not per commit, so it feels like this PR is changing a lot of things when it isn't.

**Comment by chbruyand** [comment]
> I'm wondering if it would be better to move the formatting of the files to a different pull request, so that we can focus on the (hopefully) non-impact of the type changes in this PR? In theory it should not matter but we get clang-tidy warnings for the whole PR, not per commit, so it feels like this PR is changing a lot of things when it isn't.

Agreed. I'll also stick to not format/clean anything not *required*.

**Review by rgacogne — APPROVED** [review]
The change looks good to me, there is no valid reason for the length or the offset to be negative. Of course there is a small risk than an existing piece of code relies on the existing behaviour, but then I think it's a bug that needs to be fixed. 
I just pushed a commit to a throw-away branch (https://github.com/rgacogne/pdns/tree/dnsname-test-negative) that aborts if the length or the offset is negative, let's see if it passes our CI.

**Comment by rgacogne** [comment]
> Of course there is a small risk than an existing piece of code relies on the existing behaviour, but then I think it's a bug that needs to be fixed. I just pushed a commit to a throw-away branch (https://github.com/rgacogne/pdns/tree/dnsname-test-negative) that aborts if the length or the offset is negative, let's see if it passes our CI.

The CI run was successful!

I would not mind at least one other review before merging this PR, though :-)

**Review by Habbie — APPROVED** [review]

File: pdns/dnsname.cc [hunk 1]
```diff
@@ -134,16 +135,19 @@ void DNSName::packetParser(const char* qpos, int len, int offset, bool uncompres
         throw std::range_error("Found compressed label, instructed not to follow");
 
       labellen &= (~0xc0);
-      int newpos = (labellen << 8) + *(const unsigned char*)pos;
+      size_t newpos = (labellen << 8) + *(const unsigned char*)pos;
 
       if(newpos < offset) {
-        if(newpos < (int) minOffset)
+        if(newpos < static_cast<size_t>(minOffset)) {
```
Comment (omoerbeek) [inline pdns/dnsname.cc:141]: I think this cast is not needed, as `newpos` and  and `minOffser` are both unsigned types.

**Review by omoerbeek — APPROVED** [review]
One nit

File: pdns/dnsname.cc (hunk 1 above)
Comment (chbruyand) [reply pdns/dnsname.cc]: Done, thanks.

**Pull request merged**